results-processor/wptreport.py

```diff
@@ -154,13 +196,17 @@ def write_gzip_json(filepath, payload):
 
     @property
     def results(self):
-        """The 'results' field of the report, or [] if it doesn't exists."""
-        return self._report.get('results', [])
+        """The 'results' field of the report."""
+        return self._report['results']
```

Comment:
Looks like the `.get('results', [])` was unnecessary before as well, or was it not guaranteed then that the `InsufficientDataError` bailout point had been crossed?

Reply:
No, there was no guarantee that we had have checked that and `results` might not exist. In fact, there is still no programmatic guarantee with this PR (there's only the convention that `finalize` should be called before using the data), but in that case `results` will now be an empty array by default.